### PR TITLE
Allow thread meta data to be read

### PR DESCRIPTION
**Pull request by Pezzah**
## Goal

Currently it is not possible to read information from thread meta data out of the library, it is possible that someone would want to do this as part of the logging process.

The thread meta data should also be merged before the global callbacks are run, so that they can have access to a more complete report.

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [ ] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: bugsnag/src/main/java/com/bugsnag/Bugsnag.java [hunk 1]
```diff
@@ -446,6 +446,9 @@ public boolean notify(Report report, Callback reportCallback) {
             return false;
         }
 
+        // Add thread metadata to the report
+        report.mergeMetaData(THREAD_METADATA.get());
```
Comment (simonbowring) [inline bugsnag/src/main/java/com/bugsnag/Bugsnag.java]: This will have implications on which item of metadata is used in the case of duplicate keys when merging metadata. Before the precedence was:
1. Report metadata 
1. Thread metadata
1. Global metadata

Now it will be:
1. Report metadata
1. Global metadata
1. Thread metadata

Comment (Pezzah) [reply bugsnag/src/main/java/com/bugsnag/Bugsnag.java]: That's correct. I think we should add the thread metadata first, so it can be used in any global level callbacks

**Comment by fractalwrench** [comment]
Closing this PR as I can't see a ticket for it in the Pipeline Sprint and there hasn't been any further activity. @Pezzah or @simonbowring if there was a reason we need this in the notifier let me know and I can reopen.
